### PR TITLE
User can comment on another post

### PR DESCRIPTION
**Pull request by salindholm**
[User can comment on another post](https://www.pivotaltracker.com/story/show/176613368)

## User Story
```
As an API,            
In order to receive and display the comment on posts,         
I want to be able to update the posts in index
```

## Changes proposed in this pull request:
* Adding comment model and comments controller
* Writing a test
* Adding serializers and generating IndexSerializer for Post
* Adding validations of content to comments

## What I have learned working on this feature:
* The importance of configuring the right way

## Packages/Gems added
* active_model_serializers

### CODE REVIEW

File: spec/factories/users.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 FactoryBot.define do
   factory :user do
-    email { "user@random.com" }
+    sequence(:email) { |n| "test-#{n}@example.com" }
```
Comment (sealfury) [inline spec/factories/users.rb:3]: this was the solution i was looking for the other day! nice work.

File: app/controllers/api/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    else
+      render json: { message: comment.errors.full_messages.to_sentence }, status: 422
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:content, :post_id)
+  end
```
Comment (sealfury) [inline app/controllers/api/comments_controller.rb:20]: prob need to include username or email too in the comment params

Comment (KyunginNa) [reply app/controllers/api/comments_controller.rb:20]: Good thinking! Fixed it :)

File: app/controllers/api/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class Api::CommentsController < ApplicationController
+  before_action :authenticate_user!, only: [:create]
```
Comment (ViktorHek) [inline app/controllers/api/comments_controller.rb]: you forgot the index

Comment (salindholm) [reply app/controllers/api/comments_controller.rb]: This is now fixed!

**Pull request merged**